### PR TITLE
Replace some code to separate files

### PR DESCRIPTION
**Pull request by davydovanton**
In this PR I lay the foundation of splitting all this big file to small and easy for understanding files.

### Why separated file will be better
`fileutils.rb` have 1600 LoC, after moving some code to small files we can start to refactor this code and fix some strange and hard for understanding parts. Also, support will be easy.

WDYT?

### CODE REVIEW

File: lib/fileutils/entry.rb [hunk 1]
```diff
@@ -0,0 +1,353 @@
+module FileUtils
```
Comment (amatsuda) [inline lib/fileutils/entry.rb]: Needs `# frozen_string_literal: true`

File: lib/fileutils/stream_utils.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module FileUtils
```
Comment (amatsuda) [inline lib/fileutils/stream_utils.rb]: Needs `# frozen_string_literal: true`

File: lib/fileutils/entry.rb [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+      DIRECTORY_TERM = "(?=/|\\z)"
+    end
+    SYSCASE = File::FNM_SYSCASE.nonzero? ? "-i" : ""
+
+    def descendant_directory?(descendant, ascendant)
+      /\A(?#{SYSCASE}:#{Regexp.quote(ascendant)})#{DIRECTORY_TERM}/ =~ File.dirname(descendant)
+    end
+  end
+end
+
```
Comment (amatsuda) [inline lib/fileutils/entry.rb]: ✂️

**Comment by davydovanton** [comment]
@amatsuda thanks for the review 👍 
I added freeze comment to files

**Comment by amatsuda** [comment]
I would like to ask the original author @aamine's opinion on this.

Sure, this file can be separated to several smaller files, but I'm not really sure whether this file is too big to maintain.
To me it seems kind of natural that a library dealing with low-level thing becomes this size. In fact there are dozens of bigger files in Ruby stdlibs.

**Comment by hsbt** [comment]
@davydovanton Sorry for late response. This patch is out of date at 2022. I'll close this.

**Pull request closed without merging**